dali/test/python/test_torch_pipeline_rnnt.py

```diff
@@ -269,25 +282,31 @@ def test_rnnt_data_pipeline():
     comparing to the reference: random operations (i.e. dither and presampling
     aka "speed perturbation") are turned off
     """
-    batch_size = 2
     ref_pipeline = FilterbankFeatures(sample_rate=16000, n_fft=512, highfreq=.0, dither=.00001,
                                       frame_splicing=3)
-    data_path = os.path.join(dali_extra_path, "db", "audio", "rnnt_data_pipeline")
-    rec_names = ["and_showed_itself_decoded.npy", "asked_her_father_decoded.npy"]
-    recordings = [np.load(os.path.join(data_path, name)) for name in rec_names]
-    pipe = RnntTrainPipeline(device_id=0, n_devices=1, file_root=data_path,
-                             file_list=os.path.join(data_path, "file_list.txt"),
-                             batch_size=batch_size)
+    recordings = []
+    for fpath in npy_files:
+        arr = np.load(fpath)
+        arr = _convert_samples_to_float32(arr)
+        recordings.append(arr)
+    batch_size = len(recordings)
+    pipe = RnntTrainPipeline(device_id=0, n_devices=1, files=audio_files, batch_size=batch_size)
     pipe.build()
     dali_out = pipe.run()
-    seq_len = torch.tensor([rec.shape[0] for rec in recordings])
-    reference_data = [ref_pipeline.forward(torch.tensor([rec]), seq_len) for rec in recordings]
+    reference_data = []
+    for i in range(batch_size):
+        reference_data.append(
+            ref_pipeline.forward(torch.tensor([recordings[i]]), torch.tensor([recordings[i].shape[0]]))
+        )
     for sample_idx in range(batch_size):
         output_data = dali_out[0].at(sample_idx)
         output_data = np.transpose(output_data, (1, 0))
         audio_len = dali_out[1].at(sample_idx)[0]
         assert audio_len == reference_data[sample_idx].shape[2]
-        assert reference_data[sample_idx].shape[1:] == output_data.shape
-        size = reference_data[sample_idx][1:].flatten().shape[0]
-        assert np.sum(
-                np.isclose(reference_data[sample_idx], output_data, atol=.01, rtol=0)) / size > .99
+        nfeatures = reference_data[sample_idx].shape[1]
+        assert nfeatures == output_data.shape[0]
```

Comment:
Is it equivalent to the previous implementation? Does the data has only 2-dimensions?

Reply:
yes, the first dimension is 1 in the reference (we are calculating the reference sample by sample).